File: TelerikUIExamples/TelerikUIExamples/CalendarCustomization.m
//
//  CalendarCustomization.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "CalendarCustomization.h"
#import <TelerikUI/TelerikUI.h>
#import "CustomCell.h"

@interface CalendarCustomization () <TKCalendarDelegate>

@property (nonatomic, strong) TKCalendar *calendarView;

@end

@implementation CalendarCustomization

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.calendarView = [[TKCalendar alloc] initWithFrame:CGRectZero];
    self.calendarView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    self.calendarView.delegate = self;
    [self.view addSubview:self.calendarView];
    
    UIImage *img = [UIImage imageNamed:@"calendar_header"];

    TKCalendarMonthPresenter *presenter = (TKCalendarMonthPresenter*)self.calendarView.presenter;
    presenter.style.titleCellHeight = 20;
    presenter.headerView.contentMode = UIViewContentModeScaleToFill;
    presenter.headerView.backgroundColor = [UIColor colorWithPatternImage:img];
}

- (void)viewDidLayoutSubviews
{
    [super viewDidLayoutSubviews];
    CGFloat width = fminf(CGRectGetWidth(self.exampleBounds), CGRectGetHeight(self.exampleBounds));
    self.calendarView.frame = CGRectMake(self.exampleBounds.origin.x + (CGRectGetWidth(self.exampleBounds) - width)/2., self.exampleBounds.origin.y, width, width);
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark TKCalendarDelegate

- (TKCalendarCell *)calendar:(TKCalendar *)calendar viewForCellOfKind:(TKCalendarCellType)cellType
{
    if (cellType == TKCalendarCellTypeDay) {
        return [[CustomCell alloc] init];
    }
    return nil;
}

@end

File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/CalendarCustomization.swift
//
//  CalendarCustomization.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

class CalendarCustomization: TKExamplesExampleViewController, TKCalendarDelegate {
    
    let calendarView = TKCalendar()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.calendarView.autoresizingMask = UIViewAutoresizing(rawValue: UIViewAutoresizing.FlexibleWidth.rawValue | UIViewAutoresizing.FlexibleHeight.rawValue)
        self.calendarView.delegate = self
        self.view.addSubview(self.calendarView)
        
        let img = UIImage(named: "calendar_header")
        let presenter = self.calendarView.presenter as! TKCalendarMonthPresenter
        presenter.style.titleCellHeight = 20
        presenter.headerView.contentMode = UIViewContentMode.ScaleToFill
        presenter.headerView.backgroundColor = UIColor(patternImage: img!)
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        let width = min(CGRectGetWidth(self.view.bounds), CGRectGetHeight(self.view.bounds))
        self.calendarView.frame = CGRectMake(self.view.bounds.origin.x + (CGRectGetWidth(self.view.bounds) - width)/2.0, self.view.bounds.origin.y, width, width)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
//MARK: TKCalendarDelegate
    
    func calendar(calendar: TKCalendar, viewForCellOfKind cellType: TKCalendarCellType) -> TKCalendarCell? {
        if cellType == TKCalendarCellType.Day {
            return CustomCell()
        }
        return nil
    }
}